### ISSUE TITLE
allow for markdown type research analysis

### ISSUE THREAD

**Issue by djayatillake** [lightdash/lightdash#429]
possibly an extension to dbt analyses...

would allow a user to make a document with text interspersed with tables and graphs that can be saved and shared, but also two way synced with yaml which would store this document as combination of markdown and analyses piped into graphs. 

Do we have a way to describe the configuration of a graph in Lightdash as YAML? The query itself will be a dbt analyses with refs and joins and metrics, custom dimensions/metrics too, but I mean what type of graph it is, does it have a legend, what's on the y axis and x axis, does it have values labelled for a data series etc. If not will raise a separate issue.

**Comment by djayatillake** [comment]
I would say this is low priority as not every BI tool out there allows for this, in fact relatively few. However, could be a nice feature as a selling point too.

**Comment by TuringLovesDeathMetal** [comment]
I like this idea. I think I would avoid writing these analyses in YAML though and stick to markdown. 

We already have the ability to save charts. This includes the visualization, all of its settings, and the SQL query used to generate the data for the chart. Imagine if we were able to reference saved charts in the markdown file, and when we opened these in Lightdash, they would embed the chart into the analysis. That would be dope. 

But, inevitably, someone will want to "just write SQL" and not have to try to build a chart in Lightdash to show some numeric output. 

If we could save the SQL someone wrote in the `sql runner`, this would be the same as the `analyses` in dbt...so, in an ideal world, these would actually save into the dbt `analyses` folder, and then, you could reference these in the markdown file. (just made me think: how cool would it be if you could access `dbt/analyses` in the SQL runner page? 😱 )

This setup would be tricky because we'd have to have write access to their dbt project...which is what we've been trying to avoid by creating a `lightdash` folder where all of this saved content would live...we'd need to think about this bit some more...

Another thing that I think could be a neat feature to include here: having the analysis _**not**_ rerun unless you chose to have it rerun. 

Dashboards + charts used for reporting tend to be dynamic - you want these to be updated every time you load them. Analyses are static pieces (most of the time) - you actually _**don't**_ want analyses to be updated every time you load them. E.g. if underlying data changes, you delete/rename columns, etc. you don't really want to lose an analysis over this.

I can imagine an analysis would be saved as a static file, with actual pictures of the charts. But, if you chose to rerun it, then it would recompile + rerun all of the queries, etc.

**Comment by TuringLovesDeathMetal** [comment]
There's a bunch of work we'd need to complete before this (at the minimum:[dashboards](https://lightdash.notion.site/Product-Roadmap-49e980c97bb44aa5a78ee2618851f11f?p=77280688b9da4e1694702a7aaacdb6d3) + being able to [save content as code](https://lightdash.notion.site/Product-Roadmap-49e980c97bb44aa5a78ee2618851f11f?p=7d30599d56654ecd9a0e8187634b74e3)). But I think it would be SO cool if you had all of your insights in one place. Like, if I was in Marketing and I just went to the marketing space in Lightdash and I could see the highest upvoted analyses, the most viewed dashboards, the most frequently asked questions in the marketing data models, etc. - that's the dream.

**Comment by djayatillake** [comment]
> I like this idea. I think I would avoid writing these analyses in YAML though and stick to markdown: it's way more intuitive to write in, there are loads of tools that let you preview the rendering, and you can copy-paste the format from other places like Notion.
> 
> We already have the ability to save charts. This includes the visualization, all of its settings, and the SQL query used to generate the data for the chart. Imagine if we were able to reference saved charts in the markdown file, and when we opened these in Lightdash, they would embed the chart into the analysis. That would be dope.
> 
> But, inevitably, someone will want to "just write SQL" and not have to try to build a chart in Lightdash to show some numeric output.
> 
> If we could save the SQL someone wrote in the `sql runner`, this would be the same as the `analyses` in dbt...so, in an ideal world, these would actually save into the dbt `analyses` folder, and then, you could reference these in the markdown file. (just made me think: how cool would it be if you could access `dbt/analyses` in the SQL runner page? 😱 )
> 
> This setup would be tricky because we'd have to have write access to their dbt project...which is what we've been trying to avoid by creating a `lightdash` folder where all of this saved content would live...we'd need to think about this bit some more...
> 
> Another thing that I think could be a neat feature to include here: having the analysis _**not**_ rerun unless you chose to have it rerun.
> 
> Dashboards + charts used for reporting tend to be dynamic - you want these to be updated every time you load them. Analyses are static pieces (most of the time) - you actually _**don't**_ want analyses to be updated every time you load them. E.g. if underlying data changes, you delete/rename columns, etc. you don't really want to lose an analysis over this.
> 
> I can imagine an analysis would be saved as a static file, with actual pictures of the charts. But, if you chose to rerun it, then it would recompile + rerun all of the queries, etc.

would the markdown end up in their dbt repo? Right now we use confluence for this kind of thing.

**Comment by TuringLovesDeathMetal** [comment]
That's what I'm thinking 🤔...but now I'm thinking this might not be the best option.

If we made you write analyses as markdown which were saved to your dbt repo, we'd basically be saying "if you do analyses, you should do them in Lightdash". But I don't want to try to rebuild a knowledge management platform 😅 

The experience we're trying to improve here (I think - please correct me if I've misinterpreted it!) is:
> "it's annoying to try to write the text version of an analysis somewhere, but have the calculations exist in a different place."

Some reasons for this are:
- if the calculations update, you have to re-sync or replace all of the figures and numbers you've put into your analysis document. 
- if you're reading someone's analysis, it's hard to go from the figure --> the source code (or the numbers generating that figure)
(am I missing anything?)

I'm wondering if actually being able to embed Lightdash charts into existing knowledge management tools might work as a solution instead? Or what would be missing if we did this...(until we decide to become the baddest analyses + insight knowledge management tool out there 😏 )

**Comment by djayatillake** [comment]
Yes I think you're correct about the experience we're trying to achieve.

It's good to have it in the dbt repo with refs etc because then you know if you're going to break a piece of research with a dbt change (may or may not matter depending on age of the research), also re-running it is easy.

The reasons you've listed are good too, it's interesting with the first one that the numbers may change.

I wouldn't say it's a terrible experience right now, we either do confluence and embed Looker charts, or we can do markdown inside a Looker dashboard, I was wondering if having it in the repo would mean the research was less likely to go stale and more likely to be reused in the future. I think that was the reasoning behind why dbt made analyses in the first place, the research I'm describing is just a collection of analyses displayed as charts with some markdown wrapped around.

**Comment by TuringLovesDeathMetal** [comment]
>  I was wondering if having it in the repo would mean the research was less likely to go stale and more likely to be reused in the future

^This makes a lot of sense! Cool, I'll think about this some more - but agreed with what you said at the top, I don't think this is a super high priority feature, but I'd love for us to spend some time really thinking about how Lightdash integrates with analyses (because I don't think this any BI tool has nailed this experience)

**Comment by JohnRomanski** [comment]
Bumping this issue because of the content as code work that is going on.

It's true that not all BI tools do this, Looker does not, and it's a big reason why we are seeking alternatives (https://evidence.dev has an interesting approach to this that is worth a look.)

A solution we have used in the past is a simple database that contains our notes for the weekly summaries, with a free form entry (markdown as Katie suggested above) as well as a timestamp for each record. We filter the markdown results based on the filter on the dashboard to display the text for the week - this could be a lightweight way to manage this kind of data without committing it to a repo. We also use this to capture narrative around major anomalies in the data. (At CNN: traffic spikes around elections, for e-commerce sales spikes around Black Friday)
